test/base/src/androidTest/kotlin/io/realm/test/shared/QueryTests.kt

```diff
@@ -668,34 +775,169 @@ class QueryTests {
         realm.writeBlocking {
             // Queries inside a write transaction produce live results which means they can be
             // reused within the closure
-            val countQuery = query(QuerySample::class).count()
+            val countQuery = query<QuerySample>().count()
 
             assertEquals(0, countQuery.find())
             copyToRealm(QuerySample())
             assertEquals(1, countQuery.find())
         }
 
-        realm.query(QuerySample::class)
+        realm.query<QuerySample>()
             .count()
             .find { countValue -> assertEquals(1, countValue) }
     }
 
     @Test
-    @Ignore
+    fun count_asFlow_initialValue() {
+        val channel = Channel<Long>(1)
+
+        runBlocking {
+            val observer = async {
+                realm.query<QuerySample>()
+                    .count()
+                    .asFlow()
+                    .collect { countValue ->
+                        assertNotNull(countValue)
+                        channel.send(countValue)
+                    }
+            }
+
+            assertEquals(0, channel.receive())
+
+            observer.cancel()
+            channel.close()
+        }
+    }
+
+    @Test
     fun count_asFlow() {
-        // TODO
+        val channel = Channel<Long>(1)
+
+        runBlocking {
+            val observer = async {
+                realm.query<QuerySample>()
+                    .count()
+                    .asFlow()
+                    .collect { countValue ->
+                        assertNotNull(countValue)
```

Comment:
We discussed this at some point and it is hard to have all aggregate queries under `RealmScalarQuery` since count isn't nullable per definition but max and min can return null if a table is empty.

I will quickly prototype a nullable interface and a non-nullable one instead for better type safety and see how it feels.